core/node/rpc/archiver.go

```diff
@@ -101,6 +144,82 @@ func NewArchiver(
 	return a
 }
 
+func (a *Archiver) setupStatisticsMetrics(factory infra.MetricsFactory) {
+	statsMetrics := []struct {
+		name     string
+		help     string
+		getValue func() float64
+	}{
+		{
+			"streams_examined",
+			"Total streams monitored by the archiver",
+			func() float64 { return float64(a.streamsExamined.Load()) },
+		},
+		{
+			"streams_created",
+			"Total streams allocated on disk by the archiver since the last boot",
+			func() float64 { return float64(a.streamsCreated.Load()) },
+		},
+		{
+			"streams_up_to_date",
+			"Total ArchiveStream executions that did not see stream updates",
+			func() float64 { return float64(a.streamsUpToDate.Load()) },
+		},
+		{
+			"success_ops_count",
+			"Total successful ArchiveStream executions",
+			func() float64 { return float64(a.successOpsCount.Load()) },
+		},
+		{
+			"failed_ops_count",
+			"Total failed ArchiveStream executions",
+			func() float64 { return float64(a.failedOpsCount.Load()) },
+		},
+		{
+			"miniblocks_processed",
+			"Total miniblocks downloaded and stored since the last boot",
+			func() float64 { return float64(a.miniblocksProcessed.Load()) },
+		},
+		{
+			"new_stream_allocated",
+			"Total streams allocated in response to detected stream allocation events",
+			func() float64 { return float64(a.newStreamAllocated.Load()) },
+		},
+		{
+			"stream_placement_updated",
+			"Total stream placement changes",
+			func() float64 { return float64(a.streamPlacementUpdated.Load()) },
+		},
+		{
+			"stream_last_miniblock_updated",
+			"Total miniblock update events",
+			func() float64 { return float64(a.streamLastMiniblockUpdated.Load()) },
+		},
+	}
+
+	for _, metric := range statsMetrics {
+		factory.NewGaugeFunc(
```

Comment:
but why this two-step process? just call factory.NewGaugeFunc directly instead of creating array and then iterating. it's same of amount of text, but more readable and to the point. 

(It's also fine to replace these atomics with metrics completely, i.e. inc/dec metrics directly)

also printStats, etc. can be removed - once they are published as metrics, there is no need to log them anymore.

Reply:
I think I just copied the code from postgres and didn't fully realize it didn't need to be so complex.

As discussed in towns, I'll look into using a test metrics collector so we can check the output of the metrics collection for test cases.

Reply:
eh, I think it's just easier to use the atomics.